### ISSUE TITLE
extraHOCs property might not be a best DX

### ISSUE THREAD

**Issue by yhaskell** [ArnaudBarre/eslint-plugin-react-refresh#107]
Hello there!

Although current solution of having extraHOCs solves the issues we're having, in the monorepo it makes for a little-bit of a weird situation, 
where we have to manually enable every HOC in the config and they are enabled for multiple packages (since they share the same config).

Instead, I would propose an alternative config property, for example `hocComment`.
The usage would be as follows:

```js
/* eslint.config.js */

export default defineConfig({
      extends: [reactRefresh.configs.vite],
      rules: {
        "react-refresh/only-export-components": ["error", { hocComment: "allow-hoc" }],
     }
});

/* my-amazing-component.jsx */

// react-refresh: allow-hoc
export default withUniqueHocUsage(Component);
```

That maintains backward-compatibility and also allows to explicitly mark the HOCs in the files without requiring users to change the config for every HOC in the projects.

**Comment by ArnaudBarre** [comment]
Hey sorry I forgot to answer

I think that in the eslint spirit, using an 'ignore-next-line' comment for cases like is expected
